### PR TITLE
storage: initialize source/sink statistics with zero/null values.

### PR DESCRIPTION
**Pull request by guswynn**
Closes https://github.com/MaterializeInc/materialize/issues/26829
cc @RobinClowers 

### Motivation
  * This PR adds a known-desirable feature.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/A

### CODE REVIEW

**Comment by guswynn** [comment]
Because this forces a write of zeros/null at a statistics interval, delaying future updates an additional interval, there is a chance that some nightly tests timeout because I haven't set their statistics interval to 2s like in most `test/testdrive` tests that use statistics.

@MaterializeInc/testing I'll run nightlies to make sure

**Comment by benesch** [comment]
De-requesting review unless you need me. I'm +1 on the idea, but don't have time to review the code.

File: src/storage-client/src/statistics.rs [hunk 1]
```diff
@@ -464,6 +464,26 @@ pub struct SourceStatisticsUpdate {
     pub offset_committed: Gauge<Total>,
 }
 
+impl From<GlobalId> for SourceStatisticsUpdate {
+    fn from(id: GlobalId) -> SourceStatisticsUpdate {
```
Comment (petrosagg) [inline src/storage-client/src/statistics.rs:468]: Using `From` for this feels a bit weird. It's not really that `GlobalId` is converted into a `SourceStatisticsUpdate` in the same way that a `usize` is not converted into a `Vec` when it goes through `with_capacity`. I suggest making these `new` methods

File: src/storage-client/src/statistics.rs [hunk 2]
```diff
@@ -667,6 +687,18 @@ pub struct SinkStatisticsUpdate {
     pub bytes_committed: Counter,
 }
 
+impl From<GlobalId> for SinkStatisticsUpdate {
+    fn from(id: GlobalId) -> SinkStatisticsUpdate {
```
Comment (petrosagg) [inline src/storage-client/src/statistics.rs:691]: ditto

**Review by petrosagg — APPROVED** [review]

**Comment by RobinClowers** [comment]
Exciting!

**Comment by teskje** [comment]
Thanks for doing the work @guswynn!

**Pull request merged**